auth/middleware.go

```diff
@@ -87,37 +95,53 @@ func NewMiddleware(oAuthConfig OAuthConfig, options Options) *Middleware {
 	return m
 }
 
-// Authenticate authenticates a request and returns the Token if validation was successful, otherwise error is returned
-func (m *Middleware) Authenticate(r *http.Request) (Token, error) {
+// authenticate authenticates a request and returns the Token and the client certificate if validation was successful,
+// otherwise error is returned
+func (m *Middleware) authenticate(r *http.Request) (Token, *x509.Certificate, error) {
 	// get Token from Header
 	rawToken, err := extractRawToken(r)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	token, err := m.parseAndValidateJWT(rawToken)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	const forwardedClientCertHeader = "x-forwarded-client-cert"
+	var cert *x509.Certificate
+	cert, err = parseCertString(r.Header.Get(forwardedClientCertHeader))
+	if err != nil {
+		return nil, nil, err
+	}
+	if "1" == "" && cert != nil { // TODO integrate proof of possession into middleware
```

Comment:
When should this be activated? Since the parsing is already mandatory above.

Reply:
parsing it makes sense as we need it for all proof of possession validators.